libraries/rush-sdk/src/index.ts

```diff
@@ -61,8 +63,48 @@ if (rushLibModule === undefined) {
 
 // SCENARIO 3:  A tool or script depends on "rush-sdk", and is meant to be used inside a monorepo folder.
 // In this case, we can use install-run-rush.js to obtain the appropriate rush-lib version for the monorepo.
-//
-// NOT IMPLEMENTED YET
+if (rushLibModule === undefined) {
+  try {
+    const rushJsonPath: string | undefined = tryFindRushJsonLocation(process.cwd());
+    if (!rushJsonPath) {
+      throw new Error('Could not find rush.json');
+    }
+    const monorepoRoot: string = path.dirname(rushJsonPath);
+
+    const rushJson: JsonObject = JsonFile.load(rushJsonPath);
+    const { rushVersion } = rushJson;
+
+    const installRunNodeModuleFolder: string = path.join(
+      monorepoRoot,
+      `common/temp/install-run/@microsoft+rush@${rushVersion}`
+    );
+
+    try {
+      // First, try to load the version of "rush-lib" that was installed by install-run-rush.js
+      rushLibModule = requireRushLibUnderFolderPath(installRunNodeModuleFolder);
+    } catch (e) {
+      try {
+        const installAndRunRushJSPath: string = path.join(monorepoRoot, 'common/scripts/install-run-rush.js');
+        Executable.spawnSync('node', [installAndRunRushJSPath, '--help'], {
+          stdio: 'ignore'
+        });
+
+        // Retry to load "rush-lib" after install-run-rush run
+        rushLibModule = requireRushLibUnderFolderPath(installRunNodeModuleFolder);
+      } catch (e) {
+        throw new Error(`Could not load @microsoft/rush-lib from ${installRunNodeModuleFolder}`);
+      }
+    }
+
+    if (rushLibModule !== undefined) {
+      // to track which scenario is active and how it got initialized.
+      global.___rush___rushLibModule = rushLibModule;
+    }
+  } catch (e) {
+    // no-catch
+  }
+}
+
 if (rushLibModule === undefined) {
   // This error indicates that a project is trying to import "@rushstack/rush-sdk", but the Rush engine
   // instance cannot be found.  If you are writing Jest tests for a Rush plugin, add "@microsoft/rush-lib"
```

Comment:
```ts
  // to the devDependencies for your project.
  throw new Error('The "@rushstack/rush-sdk" package context has not been initialized.');
```

This error message is only relevant to Scenario 1.  But Scenario 3 is now the most common/important error case.

Reply:
Suggested new text:

```ts
'The @rushstack/rush-sdk package was not able to load the Rush engine:'
```

And then we need to show different strings depending on what went wrong, something like:

- `'Unable to find rush.json in the current folder or its parent folders. This tool is meant to be invoked from a working directory inside a Rush repository.'`
- `'The @microsoft/rush-lib package failed to install'`
- `'The @microsoft/rush-lib package failed to load'`